src/Microsoft.SqlTools.ServiceLayer/ObjectManagement/ObjectTypes/Database/DatabaseHandler.cs

```diff
@@ -626,6 +641,61 @@ private string ConfigureDatabase(InitializeViewRequestParams viewParams, Databas
                             }
                         }
 
+                        if (!viewParams.IsNewObject)
+                        {
+                            HashSet<int> fileIdsToRemove = new HashSet<int>(prototype.Files.Select(file => file.ID));
+                            foreach (var file in database.Files)
+                            {
+                                // Add a New file
+                                if(file.Id == 0)
+                                {
+                                    DatabaseFilePrototype newFile = new DatabaseFilePrototype(dataContainer, prototype, fileTypesEnums[file.Type]);
+                                    newFile.Name = file.Name;
+                                    newFile.InitialSize = (int)file.SizeInMb;
+                                    newFile.PhysicalName = file.FileNameWithExtension;
+                                    newFile.Folder = file.Path;
+                                    newFile.DatabaseFileType = fileTypesEnums[file.Type];
+                                    newFile.Exists = false;
+                                    newFile.Autogrowth = GetAutogrowth(prototype, file);
+
+                                    // Log file do not support file groups
+                                    if (fileTypesEnums[file.Type] != FileType.Log)
+                                    {
+                                        FilegroupPrototype fileGroup = new FilegroupPrototype(prototype);
+                                        fileGroup.Name = file.FileGroup;
+                                        newFile.FileGroup = fileGroup;
+                                    }
+
+                                    // Add newFile to the prototype files
+                                    prototype.Files.Add(newFile);
+                                }
+                                // Edit file properties: updating the existed files with modified data
+                                else
+                                {
+                                    foreach(var existedFile in prototype.Files)
+                                    {
+                                        if(existedFile.ID == file.Id)
```

Comment:
You can simplify this using LINQ `FirstOrDefault` to get the file with the matching ID.